src/plugins/walletConnectToDapps/v2/components/modals/EIP155TransactionConfirmation.tsx

```diff
@@ -46,10 +47,11 @@ export const EIP155TransactionConfirmation: FC<
 
   const form = useForm<CustomTransactionData>({
     defaultValues: {
-      nonce: transaction.nonce ? convertHexToNumber(transaction.nonce).toString() : undefined,
-      gasLimit: transaction.gasLimit
-        ? convertHexToNumber(transaction.gasLimit).toString()
-        : undefined,
+      nonce: transaction?.nonce ? convertHexToNumber(transaction.nonce).toString() : undefined,
+      gasLimit:
+        transaction?.gasLimit ?? transaction?.gas
+          ? convertHexToNumber((transaction?.gasLimit ?? transaction?.gas)!).toString()
```

Comment:
Are we guaranteed to have either a `gas` or a `gasLimit`?
If not we might want to fallback to `0` or something to we don't throw.